feat: use safe securityContext as default (#267)
kedacore/charts: keda/README.md

```console
helm install keda kedacore/keda --namespace keda -f values.yaml
```

> ## securityContext default configuration

Reviewer: ```suggestion
## KEDA is secure by default
```

Author: +1

> KEDA's default configuration tries to be safest as possible, that's why the configuration is safe as default using these values:

Reviewer: What about this?
```suggestion
Our default configuration strives to be as secure as possible. Because of that, KEDA will run as non-root and be secure-by-default:
```

```yaml
podSecurityContext:
  operator:
    capabilities:
      drop:
      - ALL
    allowPrivilegeEscalation: false
    readOnlyRootFilesystem: true
  metricServer:
    capabilities:
      drop:
      - ALL
    allowPrivilegeEscalation: false

>     ## Metrics server needs to write the self-signed cert so it's not possible set this
>     # readOnlyRootFilesystem: true

Reviewer 2: ```suggestion
    readOnlyRootFilesystem: true
  volumes:
    metricsApiServer:
      extraVolumes:
      - name: keda-volume
      extraVolumeMounts:
      - name: keda-volume
        mountPath: /apiserver.local.config/certificates/
```

@JorTurFer see updated instructions for readOnlyRootFilesystem=true in [FAQ](https://keda.sh/docs/2.8/faq/)

Should this secure snippet be added to the secure-by-default values?

>     ## Metrics server needs to write the self-signed cert so it's not possible set this
>     # readOnlyRootFilesystem: true

Reviewer 3: ```suggestion
    readOnlyRootFilesystem: true
  volumes:
    metricsApiServer:
      extraVolumes:
      - name: keda-volume
      extraVolumeMounts:
      - name: keda-volume
        mountPath: /apiserver.local.config/certificates/
```

@JorTurFer see updated instructions for readOnlyRootFilesystem=true in [FAQ](https://keda.sh/docs/2.8/faq/)

Should this secure snippet be added to the secure-by-default values?

Author: WDYT @kedacore/helm-maintainers?

Author: Let's discuss this in the issue ;)
